server/src/org/apache/cloudstack/network/lb/CertServiceImpl.java

```diff
@@ -339,190 +348,182 @@ public SslCertResponse createCertResponse(SslCertVO cert, List<LoadBalancerCertM
-            if (!(c instanceof X509Certificate))
+        for (final Certificate c : certs) {
+            if (!(c instanceof X509Certificate)) {
                 throw new IllegalArgumentException("Invalid chain format. Expected X509 certificate");
+            }
 
-            X509Certificate xCert = (X509Certificate)c;
+            final X509Certificate xCert = (X509Certificate)c;
 
-            Principal subject = xCert.getSubjectDN();
-            Principal issuer = xCert.getIssuerDN();
+            xCert.getSubjectDN();
+            xCert.getIssuerDN();
 
-           anchors.add(new TrustAnchor(xCert, null));
+            anchors.add(new TrustAnchor(xCert, null));
         }
 
-        X509CertSelector target = new X509CertSelector();
+        final X509CertSelector target = new X509CertSelector();
         target.setCertificate((X509Certificate)cert);
 
         PKIXBuilderParameters params = null;
         try {
             params = new PKIXBuilderParameters(anchors, target);
             params.setRevocationEnabled(false);
             params.addCertStore(CertStore.getInstance("Collection", new CollectionCertStoreParameters(certs)));
-            CertPathBuilder builder = CertPathBuilder.getInstance("PKIX", "BC");
+            final CertPathBuilder builder = CertPathBuilder.getInstance("PKIX", "BC");
             builder.build(params);
 
-        } catch (InvalidAlgorithmParameterException e) {
+        } catch (final InvalidAlgorithmParameterException e) {
             throw new IllegalArgumentException("Invalid certificate chain", e);
-        } catch (CertPathBuilderException e) {
+        } catch (final CertPathBuilderException e) {
             throw new IllegalArgumentException("Invalid certificate chain", e);
-        } catch (NoSuchAlgorithmException e) {
+        } catch (final NoSuchAlgorithmException e) {
             throw new IllegalArgumentException("Invalid certificate chain", e);
-        } catch (NoSuchProviderException e) {
+        } catch (final NoSuchProviderException e) {
             throw new CloudRuntimeException("No provider for certificate validation", e);
         }
 
     }
 
-    public PrivateKey parsePrivateKey(String key, String password) throws IOException {
-
-        PasswordFinder pGet = null;
-
-        if (password != null)
-            pGet = new KeyPassword(password.toCharArray());
-
-        PEMReader privateKey = new PEMReader(new StringReader(key), pGet);
-        Object obj = null;
-        try {
-            obj = privateKey.readObject();
-        } finally {
-            IOUtils.closeQuietly(privateKey);
-        }
-
-        try {
-
-            if (obj instanceof KeyPair)
-                return ((KeyPair)obj).getPrivate();
-
-            return (PrivateKey)obj;
-
-        } catch (Exception e) {
-            throw new IOException("Invalid Key format or invalid password.", e);
+    public PrivateKey parsePrivateKey(final String key) throws IOException {
```

Comment:
Please consider adding a `Precondition.checkArgument` to check that `key` is not blank.